cdap-api/src/main/java/co/cask/cdap/api/workflow/AbstractWorkflow.java

```diff
@@ -126,4 +127,26 @@ protected final void addSpark(String spark) {
     Predicate<WorkflowContext> predicate) {
     return configurer.condition(predicate);
   }
+
+  /**
+   * Adds a local dataset instance to the {@link Workflow}.
+   * See {@link co.cask.cdap.api.dataset.DatasetDefinition} for details.
+   *
+   * @param datasetName name of the dataset instance
+   * @param typeName name of the dataset type
+   * @param properties dataset instance properties
+   */
+  protected final void createLocalDataset(String datasetName, String typeName, DatasetProperties properties) {
```

Comment:
What happen if the given dataset name already exist? Based on the wiki design, it seems it shouldn't matter as a new "local" dataset will always get created when the workflow start. In that case, we should document it clearly. This is different to the normal "createDataset" call, as 
  1. It will only create the dataset with the given properties if the dataset doesn't exist.
  2. The creation happen at app deployment time, while for local dataset case, creation happen at runtime.